### PR TITLE
Analysis.mixins now includes PolymerBehavior serializations

### PR DESCRIPTION
**Pull request by usergenic**
- [x] CHANGELOG.md has been updated
 - I put your `PolymerBehaviors` in your `generateAnalysis()['mixins']`
 - You can identify them by their `metadata.isPolymerBehavior=true`

Also,

noidea

So PTAL!

### CODE REVIEW

**Comment by usergenic** [comment]
Note, this fixes https://github.com/Polymer/polymer-analyzer/issues/577

**Comment by rictic** [comment]
Note that we'll have to manually mark #577 as fixed after it lands, as github only cares about text in commit messages when autoclosing.

**Comment by rictic** [comment]
(I have made that mistake like two dozen times)

File: src/test/generate_analysis_test.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ suite('generate-elements', () => {
 
   suite('generateAnalysisMetadata', () => {
 
-    suite('generatates for Document array from fixtures', () => {
+    suite('generates for Document array from fixtures', () => {
```
Comment (rictic) [inline src/test/generate_analysis_test.ts]: actually laughing out loud at "generatates"

File: src/generate-analysis.ts [hunk 2]
```diff
@@ -235,6 +250,18 @@ function serializeElementMixin(
   return metadata;
 }
 
+function serializePolymerBehaviorAsElementMixin(
+    behavior: ResolvedPolymerBehavior, packagePath: string): ElementMixin {
+  const metadata = serializeElementLike(behavior, packagePath) as ElementMixin;
+  metadata.name = behavior.className;
+  metadata.privacy = behavior.privacy;
+  metadata.metadata.isPolymerBehavior = true;
```
Comment (rictic) [inline src/generate-analysis.ts]: prefer:

```javascript
metadata.metadata = {polymer: {isBehavior: true}};
```

The idea is that an analysis object should be able to host multiple metadata sections contributed by multiple frameworks/libraries

File: src/test/generate_analysis_test.ts (hunk 1 above)
Comment (usergenic) [inline src/test/generate_analysis_test.ts]: generatates

**Comment by justinfagnani** [comment]
Hey @usergenic and @rictic what was the reasoning here? Behaviors aren't mixins and we already had a field for behaviors?

**Comment by usergenic** [comment]
- The motivation was to get all the details of Behaviors into the Analysis to support docs.
 - We decided on the `mixins` property as a way to get them in there:
   - Behaviors are akin to mixins (PRO), though not interchangeable (CON).
   - The goal of reusing `mixins` was to avoid polluting the Analysis format with top-level property for a framework-specific thing (PRO).

**Comment by TimvdLippe** [comment]
> Note that we'll have to manually mark #577 as fixed after it lands, as github only cares about text in commit messages when autoclosing.

Quick note: It also works if you put the issue number in the PR title or description. Easier to fix 😄

**Pull request merged**